### PR TITLE
xcodebuild_analyze and xcodebuild_analyze_ensure_no_results

### PR DESCRIPTION
**Pull request by WezSieTato**

### CODE REVIEW

File: lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_action.rb [hunk 1]
```diff
@@ -1,46 +1,82 @@
...
+            optional: true,
+            type: String),
+          FastlaneCore::ConfigItem.new(key: :static_analyzer,
+            env_name: "XCODEBUILD_ANALYZE_STATIC_ANALYZER",
+            description: "Run clang static analyzer if true",
+            optional: true,
+            type: Boolean),
+          FastlaneCore::ConfigItem.new(key: :output,
+            env_name: 'XCODEBUILD_ANALYZE_OUTPUT',
+            description: '- separated list of file output format (default: plist-html)',
```
Comment (arturdm) [inline lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_action.rb:67]: This description starts with a `-` while others do not (capital letter missing too). What would you say if we kept it consistent?

File: lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_ensure_no_results.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        configuration.output = "html"
+        Helper::XcodebuildAnalyzeHelper.run_analyzer(configuration)
+        input_path = configuration.workspace ? configuration.workspace : configuration.project
+        path = "#{File.dirname(input_path)}/#{configuration.output_dir}"
+        results = Dir["#{path}/**/**.html"]
+        FileUtils.rm_rf(path) if params[:prune_output]
+        if results.empty?
+          UI.success("Project is free of analyzer warnings, all good! 💪")
+        else
+          UI.user_error!("Analyzer found vulnerables in #{results.count} files!")
```
Comment (arturdm) [inline lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_ensure_no_results.rb:18]: `Analyzer found vulnerabilities`

Comment (arturdm) [inline lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_ensure_no_results.rb:18]: `files` -> `file(s)`

File: lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_ensure_no_results.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        end
+      end
+
+      def self.description
+        "Run code analyzer using xcodebuild and fail if found something"
```
Comment (arturdm) [inline lib/fastlane/plugin/xcodebuild_analyze/actions/xcodebuild_analyze_ensure_no_results.rb:23]: `(...) and fail if any vulnerability is found`

File: spec/xcodebuild_analyze_action_project_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+describe Fastlane::Actions::XcodebuildAnalyzeAction do
+  describe 'xcodebuild analyze should run xcodebuild with -project' do
+    before(:each) do
+      expect(Fastlane::Actions)
+        .to receive(:sh)
+        .with("xcodebuild -project Meniga.xcodeproj -scheme Meniga analyze")
+    end
+
+    it 'when project provided and workspace not provided' do
+      Fastlane::FastFile.new.parse("lane :test do
```
Comment (arturdm) [inline spec/xcodebuild_analyze_action_project_spec.rb:10]: What do you think about formatting it like this? 

```
Fastlane::FastFile.new.parse("
    lane :test do
      xcodebuild_analyze(project: 'Meniga.xcodeproj', scheme: 'Meniga')
    end
").runner.execute(:test)
```

File: spec/xcodebuild_analyze_ensure_no_results_spec.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        xcodebuild_analyze_ensure_no_results(project: 'Meniga.xcodeproj', scheme: 'Meniga')
+      end").runner.execute(:test)
+    end
+
+    it 'should failed when html file created under results path' do
+      File.new("fastlane/analyze_results/StaticAnalyzer/ExampleApp/report-57c09a.html", "w")
+
+      expect(Fastlane::UI)
+        .to receive(:user_error!)
+        .with("Analyzer found vulnerables in 1 files!")
```
Comment (arturdm) [inline spec/xcodebuild_analyze_ensure_no_results_spec.rb:32]: `Analyzer found vulnerabilities in 1 file(s)!` ?

File: spec/xcodebuild_analyze_action_project_spec.rb (hunk 4 above)
Comment (WezSieTato) [reply spec/xcodebuild_analyze_action_project_spec.rb]: I changed format and extract this to new method.

**Pull request merged**